### PR TITLE
Feat(eos_cli_config_gen): Add schema for local_users

### PR DESCRIPTION
**Pull request by KonikaChaurasiya-GSLab**
### Contributor Checklist

- [x] Create schema fragment matching data model described in README.md and README_v4.0.md
  - README.md is most complete with all keys. README_v4.0 includes partial data models after conversion to lists.
  - Schema fragment path is `roles/eos_cli_config_gen/schemas/schema_fragments/local_users.schema.yml`.
  - Copy line 1-5 from another schema (comments and outer type:dict).
  - Refer to [schema documentation](https://avd.sh/en/devel/docs/input-variable-validation-BETA.html) for syntax and/or use YAML Lint plugin from Redhat in VSCode.
  - Use `convert_types` on value that could be mixed type or misinterpreted like integers and numeric strings.
- [x] If the data model has been converted from wildcard dicts:
  - Add `convert_types: ['dict']` to the schema.
  - Remove `convert_dicts` from the `templates/eos/local-users.j2` and `templates/documentation/<local-users>.j2` templates.
- [x] Run `molecule converge -s build_schemas_and_docs` to update schema and documentation.
- [x] Test by running `molecule converge -s eos_cli_config_gen` and verify no errors or changes to generated configs/docs.

### Reviewer Checklist

- Reviewer 1: Claus
  - [x] Verify that data model is fully covered in the described schema. Easiest by looking at the generated documentation.
  - [x] Verify that `convert_dicts` has been removed from templates as applicable.
  - [x] Verify no changes to configs/docs on any molecule scenario
  - [x] Verify that CI pass

- Reviewer 2: Carl
  - [x] Verify that data model is fully covered in the described schema. Easiest by looking at the generated documentation.
  - [x] Verify that `convert_dicts` has been removed from templates as applicable.
  - [x] Verify no changes to configs/docs on any molecule scenario
  - [x] Verify that CI pass

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/local_users.schema.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# yaml-language-server: $schema=../../../../plugins/plugin_utils/schema/avd_meta_schema.json
+# Line above is used by RedHat's YAML Schema vscode extension
+# Use Ctrl + Space to get suggestions for every field. Autocomplete will pop up after typing 2 letters.
+type: dict
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/local_users.schema.yml:4]: I don't think all the keys are required. Please check the templates if it is not documented in the old data model.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/local_users.schema.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      type: dict
+      keys:
+        name:
+          type: str
+          required: true
+          display_name: Local-user Name
+        privilege:
+          type: int
+          required: true
+          display_name: Privilege
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/local_users.schema.yml]: Display_name is only needed if the name is different than the output of `key.replace('_',' ').title()`.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/local_users.schema.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        no_password:
+          type: bool
+          required: true
+          display_name: No Password
+          description: |
+            < true | do not configure a password for given username. sha512_password MUST not be defined for this user. >
+        ssh_key:
+          type: str
+          required: true
+          display_name: SSH-Key
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/local_users.schema.yml]:
```suggestion
  local_users:
    type: list
    display_name: Local Users
    required: true
    primary_key: name
    items:
      type: dict
      keys:
        name:
          type: str
          required: true
          display_name: Username
        privilege:
          type: int
          min: 1
          max: 15
          description: Initial privilege level with local EXEC authorization.
        role:
          type: str
          description: EOS RBAC Role to be assigned to the user such as "network-admin" or "network-operator"
        no_password:
          type: bool
          description: If set a password will not be configured for this user. "sha512_password" MUST not be defined for this user.
        sha512_password:
          display_name: SHA512 Hash of Password
          type: str
          description: Must be the hash of the password. By default EOS salts the password with the username, so the simplest is to generate the hash on an EOS device using the same username.
        ssh_key:
          display_name: SSH Key
          type: str
```

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

**Review by carlbuchmann — APPROVED** [review]
LGTM!

**Pull request merged**